src/jit/importer.cpp

```diff
@@ -682,6 +675,41 @@ inline void Compiler::impAppendStmt(GenTree* stmt, unsigned chkLevel)
 #endif
 }
 
+//------------------------------------------------------------------------
+// impAppendStmt: Add the statement to the current stmts list.
+//
+// Arguments:
+//    stmt - the statement to add.
+//
+inline void Compiler::impAppendStmt(GenTree* stmt)
+{
+    // Point 'prev' at the previous node, so that we can walk backwards.
+    stmt->gtPrev = impTreeLast;
+
+    // Append the expression statement to the list.
+    impTreeLast->gtNext = stmt;
+    impTreeLast         = stmt;
+}
+
+//------------------------------------------------------------------------
+// impExtractLastStmt: Extract the last statement from the current stmts list.
+//
+// Return Value:
+//    The extracted statement.
+//
+// Notes:
+//    It suppose that the stmt will be reinserted later.
+//
+GenTree* Compiler::impExtractLastStmt()
```

Comment:
Maybe `impRemoveLastStmt` would be a better name? Extract seems a bit ambiguous - does it remove it or does it only return it?

Reply:
I wanted to make #8059 easier to fix after that PR. In order to do that we need to distinguish when we:
1) remove and destroy a statement;
2) remove and keep a statement to reinsert in in the future.

and I think remove looks ambiguous here as well, but I am not sure here.